src/Core/Checkout/Customer/SalesChannel/LoginAsCustomerRoute.php

```diff
@@ -0,0 +1,108 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\SalesChannel;
+
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Checkout\Customer\CustomerException;
+use Shopware\Core\Checkout\Customer\Event\CustomerBeforeLoginEvent;
+use Shopware\Core\Checkout\Customer\Event\CustomerLoginEvent;
+use Shopware\Core\Checkout\Customer\Exception\CustomerNotFoundByIdException;
+use Shopware\Core\Checkout\Customer\LoginAsCustomerTokenGenerator;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\System\SalesChannel\Context\CartRestorer;
+use Shopware\Core\System\SalesChannel\ContextTokenResponse;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+#[Route(defaults: ['_routeScope' => ['store-api'], '_contextTokenRequired' => false])]
+class LoginAsCustomerRoute extends AbstractLoginAsCustomerRoute
+{
+    public const CUSTOMER_ID = 'customerId';
+
+    public const SALES_CHANNEL_ID = 'salesChannelId';
+
+    public const TOKEN = 'token';
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly EventDispatcherInterface $eventDispatcher,
+        private readonly EntityRepository $customerRepository,
+        private readonly CartRestorer $restorer,
+        private readonly LoginAsCustomerTokenGenerator $tokenGenerator
+    ) {
+    }
+
+    public function getDecorated(): AbstractLoginAsCustomerRoute
+    {
+        throw new DecorationPatternException(self::class);
+    }
+
+    #[Route(path: '/store-api/account/login/customer', name: 'store-api.account.login-as-customer', methods: ['POST'])]
+    public function loginAsCustomer(RequestDataBag $data, SalesChannelContext $context): ContextTokenResponse
```

Comment:
This is not possible, as it opens up a huge security vulnerability, the `store-api` is a public api, meaning you don't have to be logged in to use it, which means that everyone could call this route and by providing the id of some customer they would get access to their account

Reply:
as an alternative you could make it an `admin-api` route, where you have to be a logged in user to access the api (so it also should only be called from the administration in our case, but i think that fits for this use case) and please create a [separate ACL permission](https://developer.shopware.com/docs/guides/plugins/plugins/administration/add-acl-rules#additional-permissions) for this new route and functionality, so that administrators can give or remove access to this feature for each user specifically based on their assigned role.

Reply:
@keulinho 

Thank you for your feedback, the token gets validated meaning you have to provide an valid token for the customer to perform the login.

It would not be simple as just guessing the customerId, otherwise everyone could just login in.

Do you have any example for restoring the cart from the admin api to the storefront?